lib/kernel/src/net_kernel.erl

```diff
@@ -707,6 +721,25 @@ handle_call({getopts, Node, Opts}, From, State) ->
     end,
     async_reply({reply, Return, State}, From);
 
+
+handle_call(get_state, From, State) ->
+    Started = case State#state.supervisor of
+                  net_sup -> static;
+                  _ -> dynamic
+              end,
+    {NameType,Name} = case {persistent_term:get(net_kernel, undefined), node()} of
+                          {undefined, Node} ->
+                              {static, Node};
+                          {dynamic_node_name, nonode@nohost} ->
+                              {dynamic, undefined};
+                          {dynamic_node_name, Node} ->
+                              {dynamic, Node}
+                      end,
+    Return = #{started => Started,
+               name_type => NameType,
```

Comment:
Can we add a field for long/short domain name? possibly called `domain_type` that has values `short | long`?

Reply:
done